### DISCUSSION TITLE
Refresh mitmweb UI after a command has been run

### DISCUSSION THREAD

**Discussion by malwareinfosec** [mitmproxy/mitmproxy#6973] [Q&A] [answered]
Hi, I'm discovering the commands in mitmweb and started implementing one in my addon that could search for a string within each flow's response body (I believe this is not possible right now). With this command, it works well at searching and finding the flow, but the UI is not refreshed.

Let me show you:

This is the code:
![image](https://github.com/mitmproxy/mitmproxy/assets/25351665/e05c3e46-6736-431d-8db4-133c9403c8fc)

Then I invoke the command by typing it. However, the web UI does not reflect the changes. These should be marking each flow with a red dot and adding a comment:
![image](https://github.com/mitmproxy/mitmproxy/assets/25351665/7fa5f704-1af4-40c2-99c0-b2f3b6d28527)

I need to manually hit the refresh button of my browser to show the changes:
![image](https://github.com/mitmproxy/mitmproxy/assets/25351665/c97646d7-01d0-4b77-91ff-080224e0ac10)

Is there a way to force a refresh from within my addon?

**Comment by Prinzhorn** [comment, marked as answer]
> search for a string within each flow's response body (I believe this is not possible right now)

see https://docs.mitmproxy.org/stable/concepts-filters/ in particular `~bs`

As for the original question, I don't know

Edit: quick peek at other commands, you need to use `ctx.master.addons.trigger(hooks.UpdateHook(flows))` to notify of changes to the flows you've made, example:

https://github.com/mitmproxy/mitmproxy/blob/27f990203a6db31b5b95d703e9d7f4fb6380183b/mitmproxy/addons/core.py#L77

**Reply by mhils** [reply]
Body filters do exist and work, but currently not in mitmweb. :)

**Reply by malwareinfosec** [reply]
The filter option ``~bs`` does not seem to be supported for mitmweb (this would be a very useful feature IMO):
![image](https://github.com/mitmproxy/mitmproxy/assets/25351665/a7638e65-5beb-4188-8cb9-1d6cf5ce370f)

Thanks for the UpdateHook command! I was able to implement it.
![image](https://github.com/mitmproxy/mitmproxy/assets/25351665/12e01d17-7cb7-45b6-9049-99b84556b5a0)
